### ISSUE TITLE
Exception thrown when swift_mailer send function fails

### ISSUE THREAD

**Issue by umar456** [Kitware/CDash#1083]
An exception is thrown by CDash when the mailer fails to send an email out. The stack trace looks like this:

```
cdash_1              | [2020-10-27 05:30:26] production.INFO: Expected response code 220 but got an empty response {"function":"CDash\\Messaging\\Notification\\Email\\Mail::send"} 
cdash_1              | [2020-10-27 05:30:26] production.ERROR: Undefined variable: status {"exception":"[object] (ErrorException(code: 0): Undefined variable: status at /home/kitware/cdash/app/cdash/include/Messaging/Notification/Email/Mail.php:144)
cdash_1              | [stacktrace]
cdash_1              | #0 /home/kitware/cdash/app/cdash/include/Messaging/Notification/Email/Mail.php(144): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8, 'Undefined varia...', '/home/kitware/c...', 144, Array)
cdash_1              | #1 /home/kitware/cdash/app/cdash/include/Messaging/Notification/Mailer.php(54): CDash\\Messaging\\Notification\\Email\\Mail->send(Object(CDash\\Messaging\\Notification\\Email\\EmailMessage))
cdash_1              | #2 /home/kitware/cdash/app/cdash/include/Messaging/Notification/Mailer.php(40): CDash\\Messaging\\Notification\\Mailer->sendNotification(Object(CDash\\Messaging\\Notification\\Email\\EmailMessage))
cdash_1              | #3 /home/kitware/cdash/app/cdash/include/sendemail.php(621): CDash\\Messaging\\Notification\\Mailer::send(Object(CDash\\Messaging\\Notification\\NotificationCollection))
cdash_1              | #4 /home/kitware/cdash/app/cdash/include/do_submit.php(179): sendemail(Object(BuildHandler), 1)
cdash_1              | #5 /home/kitware/cdash/app/cdash/public/submit.php(146): do_submit(Resource id #499, 1, 408, '1304985ac35e5e5...', true)
cdash_1              | #6 /home/kitware/cdash/app/Http/Controllers/CDash.php(161): require('/home/kitware/c...')
cdash_1              | #7 /home/kitware/cdash/app/Http/Controllers/CDash.php(217): App\\Http\\Controllers\\CDash->getRequestContents()
cdash_1              | #8 /home/kitware/cdash/app/Http/Controllers/CDash.php(56): App\\Http\\Controllers\\CDash->handleSubmission()
cdash_1              | #9 [internal function]: App\\Http\\Controllers\\CDash->__invoke('submit.php')
cdash_1              | #10 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): call_user_func_array(Array, Array)
cdash_1              | #11 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction('__invoke', Array)
cdash_1              | #12 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Route.php(219): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(App\\Http\\Controllers\\CDash), '__invoke')
cdash_1              | #13 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Route.php(176): Illuminate\\Routing\\Route->runController()
cdash_1              | #14 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Router.php(680): Illuminate\\Routing\\Route->run()
cdash_1              | #15 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #16 /home/kitware/cdash/app/Http/Middleware/CheckPassword.php(29): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #17 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): App\\Http\\Middleware\\CheckPassword->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #18 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #19 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #20 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(75): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #21 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #22 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #23 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #24 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(56): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #25 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #26 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #27 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #28 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(66): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #29 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #30 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #31 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Router.php(682): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
cdash_1              | #32 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Router.php(657): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
cdash_1              | #33 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
cdash_1              | #34 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Routing/Router.php(612): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
cdash_1              | #35 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(176): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
cdash_1              | #36 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #37 /home/kitware/cdash/app/Http/Middleware/CheckDirectoryPermissions.php(69): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #38 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): App\\Http\\Middleware\\CheckDirectoryPermissions->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #39 /home/kitware/cdash/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #40 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Fideloper\\Proxy\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #41 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #42 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #43 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #44 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #45 /home/kitware/cdash/app/Http/Middleware/CheckDatabaseConnection.php(33): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #46 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): App\\Http\\Middleware\\CheckDatabaseConnection->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #47 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(62): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #48 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode->handle(Object(Illuminate\\Http\\Request), Object(Closure))
cdash_1              | #49 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
cdash_1              | #50 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(151): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
cdash_1              | #51 /home/kitware/cdash/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
cdash_1              | #52 /home/kitware/cdash/public/index.php(55): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
cdash_1              | #53 {main}
cdash_1              | "}
```
Looks like the status variable is not defined outside of the try block.

**Comment by zackgalbreath** [comment]
I think you're right. Thanks for letting us know. I'll issue a fix shortly.

**Comment by zackgalbreath** [comment]
Fixed by #1096
